### PR TITLE
CompileStatic for copy and exec too

### PR DESCRIPTION
**Pull request by advayDev1**
These are intentionally 2 commits so you can see what is required to reinstate mocking.

Let me know what you think, and I'll clean it up for real merging.

### CODE REVIEW

**Comment by advayDev1** [comment]
btw, with this, autocomplete works for the configure closures too
why Gradle does not mark all their arguments with these kinds of Groovy annotations is beyond me as it is in the Groovy language intro.  or maybe by Gradle jars haven't retained the annotations or something.

**Comment by advayDev1** [comment]
if this looks fine, i'll redo all the project.copy to use a similar wrapper method that hides the casting as well.

**Comment by advayDev1** [comment]
@brunobowden  - you don't need to review the style right now.  just look at the functionality of the PR's overall diff - not the first commit.  if the high level change will mesh well with your changes, i will clean up for actual review.

**Comment by advayDev1** [comment]
@brunobowden  - now ready for real review

File: src/main/groovy/com/github/j2objccontrib/j2objcgradle/tasks/TestTask.groovy [hunk 1]
```diff
@@ -106,7 +105,7 @@ class TestTask extends DefaultTask {
                     "    testJ2objc org.junit.runner.JUnitCore\n" +
                     "Copy the command from \"Command:\" onwards, then try varying the command\n" +
                     "to drop tests and figure out which ones are causing the failures.\n"
-                    "Then disable them as described above.\n"
+            "Then disable them as described above.\n"
```
Comment (brunobowden) [inline src/main/groovy/com/github/j2objccontrib/j2objcgradle/tasks/TestTask.groovy]: Lost indentation

File: src/main/groovy/com/github/j2objccontrib/j2objcgradle/tasks/Utils.groovy [hunk 2]
```diff
@@ -15,16 +15,25 @@
...
 import org.gradle.api.InvalidUserDataException
 import org.gradle.api.Project
+import org.gradle.api.file.CopySpec
 import org.gradle.api.file.FileCollection
 import org.gradle.api.file.SourceDirectorySet
 import org.gradle.api.plugins.JavaPluginConvention
 import org.gradle.api.tasks.SourceSet
+import org.gradle.api.tasks.WorkResult
+import org.gradle.process.ExecResult
+import org.gradle.process.ExecSpec
```
Comment (brunobowden) [inline src/main/groovy/com/github/j2objccontrib/j2objcgradle/tasks/Utils.groovy]: Remove this blankline?

Comment (advayDev1) [inline src/main/groovy/com/github/j2objccontrib/j2objcgradle/tasks/Utils.groovy]: i think the separation of java.* and non-java is normal?

File: src/main/groovy/com/github/j2objccontrib/j2objcgradle/tasks/TestTask.groovy (hunk 1 above)
Comment (advayDev1) [inline src/main/groovy/com/github/j2objccontrib/j2objcgradle/tasks/TestTask.groovy]: done

Comment (advayDev1) [inline src/main/groovy/com/github/j2objccontrib/j2objcgradle/tasks/TestTask.groovy]: lol autoformat caught a bug - the last line lacks a '+' so the message ended early.

File: src/main/groovy/com/github/j2objccontrib/j2objcgradle/tasks/XcodeTask.groovy [hunk 3]
```diff
@@ -106,9 +105,11 @@ class XcodeTask extends DefaultTask {
                 "s.resources = '${j2objcResourceDirName}/**/*'\n" +
                 "s.requires_arc = true\n" +
                 "s.preserve_paths = '${srcGenDirRelativeToBuildDir}**/*.a'\n" +
-                "s.libraries = 'ObjC', 'guava', 'javax_inject', 'jre_emul', 'jsr305', 'z', 'icucore', '${project.name}-j2objc'\n" +
+                "s.libraries = " +
+                "'ObjC', 'guava', 'javax_inject', 'jre_emul', 'jsr305', 'z', 'icucore', '${project.name}-j2objc'\n" +
```
Comment (advayDev1) [inline src/main/groovy/com/github/j2objccontrib/j2objcgradle/tasks/XcodeTask.groovy]: these changes are for 100-char limit only

File: src/main/groovy/com/github/j2objccontrib/j2objcgradle/tasks/XcodeTask.groovy [hunk 4]
```diff
@@ -106,9 +105,11 @@ class XcodeTask extends DefaultTask {
...
                 "s.xcconfig = { 'HEADER_SEARCH_PATHS' => '${getJ2objcHome()}/include', " +
```
Comment (brunobowden) [inline src/main/groovy/com/github/j2objccontrib/j2objcgradle/tasks/XcodeTask.groovy]: Put this on a separate line for clarity:

    'HEADER_SEARCH_PATHS' => '${getJ2objcHome()}/include', " +

File: src/main/groovy/com/github/j2objccontrib/j2objcgradle/tasks/Utils.groovy (hunk 2 above)
Comment (brunobowden) [inline src/main/groovy/com/github/j2objccontrib/j2objcgradle/tasks/Utils.groovy]: I think you're right too. Leave as is.

**Comment by brunobowden** [comment]
LGTM along with addressing final comments

File: src/main/groovy/com/github/j2objccontrib/j2objcgradle/tasks/XcodeTask.groovy (hunk 4 above)
Comment (advayDev1) [inline src/main/groovy/com/github/j2objccontrib/j2objcgradle/tasks/XcodeTask.groovy]: done

**Comment by advayDev1** [comment]
done and merging

**Pull request merged**